testng-core/src/test/java/test/hook/HookableTest.java

```diff
@@ -81,10 +91,23 @@ public void hookFailure() {
     TestResultsCollector listener = new TestResultsCollector();
     tng.addListener(listener);
     tng.run();
-    assertThat(listener.getPassedMethodNames()).isNotNull();
+    assertThat(listener.getPassedMethodNames()).isEmpty();
+    assertThat(listener.getFailedTests()).hasSize(1);
+    ITestResult failedTestResult = listener.getFailedTests().get(0);
+    assertThat(failedTestResult.getAttribute(HOOK_INVOKED_ATTRIBUTE)).isNotNull();
+    assertThat(failedTestResult.getAttribute(HOOK_METHOD_INVOKED_ATTRIBUTE)).isNull();
+  }
+
+  @Test
+  public void hookFailureWithStatusAltered() {
+    TestNG tng = create(HookFailureWithStatusAlteredSample.class);
+    TestResultsCollector listener = new TestResultsCollector();
+    tng.addListener(listener);
+    tng.run();
+    assertThat(listener.getPassedMethodNames()).hasSize(1);
```

Comment:
Maybe we should assert on invoked methods too.

Reply:
Fixed